### PR TITLE
Add Session workflows documentation

### PR DESCRIPTION
**Pull request by andreyvelich**
I added create and delete Session workflow to architecture diagrams.
Please take a look at the doc changes.

cc @Zsailer @kevin-bates @blink1073

### CODE REVIEW

File: docs/source/developers/architecture.rst [hunk 1]
```diff
@@ -71,3 +71,95 @@ Jupyter Server contains the following components:
   - **Kernel Spec Manager** parses files with JSON specification for a Kernels,
     and provides a list of available Kernel configurations. To learn about
     Kernel Spec, check `the Jupyter Client guide <https://jupyter-client.readthedocs.io/en/stable/kernels.html#kernel-specs>`_.
+
+Create Session Workflow
+-----------------------
+
+The create Session workflow can be seen in figure below:
```
Comment (kevin-bates) [inline docs/source/developers/architecture.rst:78]:
```suggestion
The create Session workflow can be seen in the figure below:
```

File: docs/source/developers/architecture.rst [hunk 2]
```diff
@@ -71,3 +71,95 @@ Jupyter Server contains the following components:
...
+
+.. image:: ../images/session-create.drawio.png
+   :alt: Create Session Workflow
+   :width: 90%
+   :align: center
+
+When a user starts a new Kernel, the following steps process:
```
Comment (kevin-bates) [inline docs/source/developers/architecture.rst:85]:
```suggestion
When a user starts a new kernel, the following steps occur:
```

File: docs/source/developers/architecture.rst [hunk 3]
```diff
@@ -71,3 +71,95 @@ Jupyter Server contains the following components:
...
+
+#. **Mapping Kernel Manager** starts the Kernel shutdown process by using
+   **Multi Kernel Manager** and **Kernel Manager**.
+
+#. **Kernel Manager** receives interrupt mode from **Kernel Spec Manager**.
+   **Kernel Manager** supports ``Signal`` and ``Message`` interrupt modes.
+   By default, the ``Signal`` interrupt mode is being used.
+
+   - When interrupt mode is ``Signal``, **Kernel Provisioner** kills the Kernel
+     with ``SIGINT`` operating system signal.
```
Comment (kevin-bates) [inline docs/source/developers/architecture.rst:153]:
```suggestion
   - When the interrupt mode is ``Signal``, the **Kernel Provisioner** interrupts the kernel
     with the ``SIGINT`` operating system signal (although other provisioner implementations may use a different approach).
```

Comment (kevin-bates) [reply docs/source/developers/architecture.rst:153]: The diagram for the signal-based interrupt should probably replace `Kill` with `Interrupt` as well.

File: docs/source/developers/architecture.rst (hunk 2 above)
Comment (kevin-bates) [reply docs/source/developers/architecture.rst:85]: I think "Kernel" should be "kernel" except when it's a proper noun (e.g., Mapping Kernel Manager, Kernel Spec Manager, etc.).  I believe this was also true in the general architecture section as well.  Perhaps a sweep can be made for such instances.

File: docs/source/developers/architecture.rst [hunk 4]
```diff
@@ -71,3 +71,95 @@ Jupyter Server contains the following components:
...
+
+   - When interrupt mode is ``Message``, **Session** sends
+     the ``interrupt_request`` message on the control channel. Check the
+     `messaging in Jupyter guide <https://jupyter-client.readthedocs.io/en/latest/messaging.html#msging-interrupt>`_
+     to know more about it.
+
+#. When Kernel is shutdown, **Session Manager** deletes the Session data from
+   the SQLite3 DataBase and responses 204 Status Code to the Notebook client.
```
Comment (kevin-bates) [inline docs/source/developers/architecture.rst:161]: Should the kernel's shutdown be detailed similar to its interrupt?  I know that can be a lot and we might be trying to limit text.  Perhaps we could do something like...

When the [kernel is shutdown](https://github.com/jupyter/jupyter_client/blob/00987a97b0b52431287c60060c98518cf36072f1/jupyter_client/manager.py#L467-L480), ...

and point to the code (or documentation produced from the code)?

**Review by kevin-bates — COMMENTED** [review]
Another useful diagram - thank you @andreyvelich!  Here are some initial comments...

File: docs/source/developers/architecture.rst (hunk 2 above)
Comment (andreyvelich) [reply docs/source/developers/architecture.rst:85]: Sounds good, I will change it to "kernel".

File: docs/source/developers/architecture.rst (hunk 4 above)
Comment (andreyvelich) [reply docs/source/developers/architecture.rst]: @kevin-bates I think that makes sense. I added steps after kernel interrupt.
Please let me know, what do you think.

File: docs/source/developers/architecture.rst [hunk 5]
```diff
@@ -36,38 +36,140 @@ Jupyter Server contains the following components:
...
+
+#. **Kernel Manager** waits for the kernel shutdown. After the timeout,
+   **Kernel Manager** interrupts the kernel with ``SIGTERM`` operating system signal.
+
+#. **Kernel Manager** clean up the kernel resources. It removes kernel's interprocess
+   communication ports, closes control socket, and releases Shell, IOPub, StdIn,
+   Control, and Heartbeat ports.
+
+#. When shutdown is finished, **Session Manager** deletes the Session data from
+   the SQLite3 DataBase and responses 204 Status Code to the Notebook client.
```
Comment (kevin-bates) [inline docs/source/developers/architecture.rst:171]:
```suggestion
   the SQLite3 database and responses 204 status code to the Notebook client.
```

File: docs/source/developers/architecture.rst [hunk 6]
```diff
@@ -36,38 +36,140 @@ Jupyter Server contains the following components:
...
+     (although other provisioner implementations may use a different approach).
+
+   - When interrupt mode is ``Message``, Session sends
+     the `"interrupt_request" <https://jupyter-client.readthedocs.io/en/latest/messaging.html#msging-interrupt>`_
+     message on the control channel.
+
+#. After interrupting kernel, Session sends the `"shutdown_request" <https://jupyter-client.readthedocs.io/en/latest/messaging.html#kernel-shutdown>`_
+   message on the control channel.
+
+#. **Kernel Manager** waits for the kernel shutdown. After the timeout,
```
Comment (kevin-bates) [inline docs/source/developers/architecture.rst:163]:
```suggestion
#. **Kernel Manager** waits for the kernel shutdown. After the timeout, and if it detects the kernel process is still running, the 
```

**Comment by andreyvelich** [comment]
Thank you for the review @kevin-bates!
I made these changes.

File: docs/source/developers/architecture.rst [hunk 7]
```diff
@@ -36,38 +36,145 @@ Jupyter Server contains the following components:
...
+   based on the input data.
+
+#. **Session Manager** sends kernel path to **Mapping Kernel Manager**.
+
+#. **Mapping Kernel Manager** starts the kernel create process by using
+   **Multi Kernel Manager** and **Kernel Manager**. You can learn more about
+   **Multi Kernel Manager** in
+   `the Jupyter Client APIs <https://jupyter-client.readthedocs.io/en/latest/api/manager.html#multikernelmanager-controlling-multiple-kernels>`_.
+
+#. **Kernel Manager** uses provisioner layer to launch a new kernel.
```
Comment (blink1073) [inline docs/source/developers/architecture.rst:101]:
```suggestion
#. **Kernel Manager** uses the provisioner layer to launch a new kernel.
```

**Review by kevin-bates — DISMISSED** [review]
Looks good @andreyvelich  - thank you!

**Comment by andreyvelich** [comment]
Thank you for your feedback @blink1073 @kevin-bates !

**Review by blink1073 — APPROVED** [review]
Thanks!  Failure is unrelated

**Pull request merged**